### DISCUSSION TITLE
"Enable Detect", "Enable Recording" and "Enable Snapshots" buttons in live view

### DISCUSSION THREAD

**Discussion by SeanDS** [blakeblackshear/frigate#14313] [Ask A Question]
This is probably a simple question but I can't find anything in the documentation about it. I've got Frigate set up to detect and record and it does so. However, in the live view I see these buttons which by default are disabled, which have tooltips "Enable Detect", "Enable Recording" and "Enable Snapshots". Clicking them on doesn't appear to do anything; for example, objects are still recorded whether the first button is on or off. So what are these for?

![image](https://github.com/user-attachments/assets/bac00173-a73d-4518-a7e4-2cd2c7fdb961)

**Comment by hawkeye217** [comment]
When you click on it, does it change to a blue icon without the line through it?

If not, we'll need to see your frigate logs and config file in order to be able to help.

**Reply by SeanDS** [reply]
The buttons do work in the sense they toggle when clicked:

![image](https://github.com/user-attachments/assets/70aff538-f021-465e-b7dd-056c9f8b04bc)

I want to know what the buttons actually do.

**Reply by hawkeye217** [reply]
They enable/disable detection, recording, and snapshots for that particular camera.

**Reply by SeanDS** [reply]
Hmm, I could have sworn these buttons were disabled on all three of my cameras yet I was still getting recorded detections. I will monitor this and report back if it's not the case. Thanks

**Reply by hawkeye217** [reply]
When you click the buttons, check your Frigate logs. They should report a line showing what state the camera is in after you click it.

**Reply by Sandman-wr** [reply]
Checked the logs, in fact did enable detection after making it blue.
Not to necro a thread, but what can we do to make sure the buttons are enabled at startup?

**Reply by NickM-27** [reply]
Enable detection in the config

**Reply by Sandman-wr** [reply]
```cameras:
  driveway:
    ffmpeg:
      inputs:
        - path: 
            rtsp://XXXXXXXXXXX.74:554/cam/realmonitor?channel=1&subtype=1&unicast=true&proto=Onvif
          roles:
            - detect
        - path: 
            rtsp://XXXXXXXXXXX.74:554/cam/realmonitor?channel=1&subtype=0&unicast=true&proto=Onvif
          roles:
            - record
    detect:
      width: 640
      height: 360
      fps: 5
    record:
      enabled: true
      continuous:
        days: 3
      motion:
        days: 1
    snapshots:
      enabled: true
    objects:
      track:
        - person
        - car
        - cat
        - dog
```
This is my config on each camera.
First stream is high res,  second is low res.
All cameras record -except- this one. ?

**Reply by hawkeye217** [reply]
You need to add `enabled: true` to the `detect` section, or add 

```
detect:
  enabled: true
```

at the global level.

**Reply by Sandman-wr** [reply]
Define global level for me.  You mean at the top, above all the cameras, in config?  Or in docker compose?

**Reply by hawkeye217** [reply]
At the top of your config, at the same level as the `cameras` key. See the [reference config](https://docs.frigate.video/configuration/reference/) in the official docs.

**Reply by Sandman-wr** [reply]
Ok, thanks.  I'll study more on that.  I'm still learning this, only having it run maybe a week, so figuring things out.
Thanks again.
